pkgs/by-name/ta/tachidesk-server/package.nix

```diff
@@ -0,0 +1,51 @@
+{ lib
+, stdenvNoCC
+, fetchurl
+, makeWrapper
+, jdk17_headless
+, nixosTests
+}:
+
+let
+  jdk = jdk17_headless;
+in
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "tachidesk-server";
+  version = "0.7.0";
+  revision = 1197;
+
+  src = fetchurl {
+    url = "https://github.com/Suwayomi/Tachidesk-Server/releases/download/v${finalAttrs.version}/Tachidesk-Server-v${finalAttrs.version}-r${toString finalAttrs.revision}.jar";
+    hash = "sha256-4DO1WiBCu/8ypFgJdBmEwQXQ1xaWAlbt8N5TELomVVA=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  buildCommand = ''
+    makeWrapper ${jdk}/bin/java $out/bin/tachidesk-server \
+      --add-flags "-Dsuwayomi.tachidesk.config.server.initialOpenInBrowserEnabled=false -jar $src"
+  '';
```

Comment:
Just making sure, was there a good reason for the choice of `buildCommand` over `buildPhase`? This seems to disable all other phases, making overrides with extensions to `patchPhase` or `buildPhase` completely useless without warning.

Quoting the [nixpkgs manual about stdenv phases](https://nixos.org/manual/nixpkgs/unstable/#sec-stdenv-phases):

> This generic command (mkDerivation) either invokes a script at buildCommandPath, or a buildCommand, or a number of phases. Package builds are split into phases to make it easier to override specific parts of the build (e.g., unpacking the sources or installing the binaries).

Reply:
> Just making sure, was there a good reason for the choice of `buildCommand` over `buildPhase`? This seems to disable all other phases, making overrides with extensions to `patchPhase` or `buildPhase` completely useless without warning.
> 
> Quoting the [nixpkgs manual about stdenv phases](https://nixos.org/manual/nixpkgs/unstable/#sec-stdenv-phases):
> 
> > This generic command (mkDerivation) either invokes a script at buildCommandPath, or a buildCommand, or a number of phases. Package builds are split into phases to make it easier to override specific parts of the build (e.g., unpacking the sources or installing the binaries).

As I didn't know very well what to do, I copied the [komga package](https://github.com/NixOS/nixpkgs/blob/301f4bfbd7c7d5a7cec710f91506c6eda4d32f94/pkgs/servers/komga/default.nix#L22) which seemed really close to me. I'm not sure this is the best option to choose

Reply:
Then I think it would be better if you wrote it like this (same goes for komga, but let's take that another time):

```
# I think you might have to do this, so it doesn't try to unzip the jar
dontUnpack = true;

buildPhase = ''
  runHook preBuild
  
  makeWrapper ${jdk}/bin/java $out/bin/tachidesk-server \
    --add-flags "-Dsuwayomi.tachidesk.config.server.initialOpenInBrowserEnabled=false -jar $src"
  
  runHook postBuild
'';
```